test/test_broker/iaas.GcpClient.spec.js

```diff
@@ -503,16 +561,91 @@ describe('iaas', function () {
 
     describe('#ComputeOperations', function () {
       let sandbox, client;
-      before(function () {
+      const diskName = 'disk-sample';
+      const zone = 'zone';
+      const snapshotName = 'snappy';
+      beforeEach(function () {
         sandbox = sinon.sandbox.create();
         client = new GcpClient(settings);
         sandbox.stub(GcpCompute.prototype, 'snapshot').returns(snapshotStub);
-      });
-      after(function () {
-        sandbox.restore();
+        sandbox.stub(GcpCompute.prototype, 'zone').returns(zoneStub);
+        sandbox.stub(GcpClient.prototype, 'getRandomDiskId').returns(diskName);
       });
       afterEach(function () {
         deleteSnapshotEventHandlers = {};
+        sandbox.restore();
+      });
+
+      it('disk metadata should fail if get fails', function () {
+        sandbox.restore();
+        sandbox.stub(GcpCompute.prototype, 'zone').returns(zoneFailedStub);
+        return client.getDiskMetadata(diskName, zone)
+          .catch(err => {
+            expect(err.message).to.equal('diskFailed');
+          });
+      });
+
+      it('disk metadata should be fetched successfully', function () {
+        return client.getDiskMetadata(diskName, zone).then(disk => {
+          expect(disk.volumeId).to.equal(diskName);
+          expect(disk.zone).to.equal('europe-west1-c');
+          expect(disk.type).to.equal('pd-ssd');
+          expect(disk.size).to.equal('4');
+        });
+      });
+
+      it('disk should be created successfully', function () {
+        let createDiskPromise = client.createDiskFromSnapshot(snapshotName, zone);
+        return Promise.delay(CONNECTION_WAIT_SIMULATED_DELAY)
+          .then(() => {
+            if (createDiskEventHandlers.complete && _.isFunction(createDiskEventHandlers.complete)) {
+              return createDiskEventHandlers.complete.call(createDiskEventHandlers.complete);
+            } else {
+              throw new Error('Event handlers not registered in create disk from snapshot');
+            }
+          })
+          .then(() => {
+            return createDiskPromise
+              .then(disk => {
+                expect(disk.volumeId).to.equal(diskName);
+                expect(disk.zone).to.equal('europe-west1-c');
+                expect(disk.type).to.equal('pd-ssd');
+                expect(disk.size).to.equal('4');
```

Comment:
Same as above, please verify all keys in response.

Reply:
Done in latest commit